enable/tools/apptools/commands.py

```diff
@@ -0,0 +1,236 @@
+# LICENSE.txt
+#
+
+"""
+Enable Commands
+===============
+
+This module provides :py:class:`apptools.undo.abstract_command.AbstractCommand`
+subclasses for common component manipulations, such as moving, resizing and
+setting attribute values.
+
+"""
+
+from __future__ import (division, absolute_import, print_function,
+                        unicode_literals)
+
+from apptools.undo.api import AbstractCommand
+from traits.api import Bool, Instance, Tuple, Unicode
+from traits.util.camel_case import camel_case_to_words
+
+from enable.component import Component
+
+class ComponentCommand(AbstractCommand):
+    """ Abstract command which operates on a Component """
+
+    #: The component the command is being performed on.
+    component = Instance(Component)
+
+    #: An appropriate name for the component that can be used by the command.
+    #: The default is the class name, split into words.
+    component_name = Unicode
+
+    def __init__(self, component, **traits):
+        super(ComponentCommand, self).__init__(component=component, **traits)
+
+    #-------------------------------------------------------------------------
+    # traits handlers
+    #-------------------------------------------------------------------------
+
+    def _component_name_default(self):
+        if self.component is not None:
+            return camel_case_to_words(self.component.__class__.__name__)
+        return ''
+
+
+class ResizeCommand(ComponentCommand):
+    """ Command for resizing a component
+
+    This handles the logic of moving a component and merging successive moves.
+    This class provides  ``_change_rectangle`` and ``_merge_data`` methods that
+    subclasses can override to change the move behaviour in a uniform way for
+    undo and redo operations.
+
+    """
+
+    #: The new rectangle of the component as a tuple (x, y, width, height).
+    data = Tuple
+
+    #: The old rectangle of the component as a tuple (x, y, width, height).
+    previous_rectangle = Tuple
+
+    #: whether additional resizes can be merged or if the resize is finished.
+    mergeable = Bool
+
+    def __init__(self, component, new_rectangle=None, previous_rectangle=None,
+                 **traits):
+        if previous_rectangle is None:
+            previous_rectangle = tuple(component.position)+tuple(component.bounds)
+
+        if new_rectangle is None:
+            if 'data' in traits:
+                data = traits.pop('data')
+            else:
+                raise TypeError("MoveCommand __init__ method requires "
+                                "'new_rectangle' argument.")
+        else:
+            data = new_rectangle
+
+        super(ComponentCommand, self).__init__(
+            component=component,
+            data=data,
+            previous_rectangle=previous_rectangle,
+            **traits)
+
+    @classmethod
+    def move_command(cls, component, new_position, previous_position=None,
```

Comment:
Because it creates a `ResizeCommand` which actually performs a move operation (ie. doesn't change the bounds, only the position.